docs/abac-rocksdb-store-format.md

```diff
@@ -0,0 +1,202 @@
+4. **Triple Encoding**: A triple (subject, predicate, object) is encoded by individually encoding each node in the triple as described above.
+
+#### Node - Type Identification
+The encoding process begins by identifying the node type using the NodeType class. 
+The node can be of type:
+- **URI**: A Uniform Resource Identifier.
+- **Literal**: A value or string.
+- **Blank**: An anonymous node.
+- **Any**: A wildcard node, represented by the asterisk (*) character.
+
+#### Node - String Representation
+Depending on the node type, the node is converted into its corresponding string representation:
+- **URI**: Encoded directly as a string.
+- **Literal**: Encoded with an appropriate prefix or suffix to distinguish it.
+- **Blank**: Encoded with a prefix indicating its anonymous nature.
+- **Any**: Simply represented by an asterisk (*).
+
+#### Node - String Formatting
+The node’s string representation is then formatted to include a type marker. For example:
+- **URI**: "u:<URI_String>".
+- **Literal**: "l:<Literal_String>".
+- **Blank**: "b:<Blank_Node_String>".
+- **Any**: '*'.
+
+### Node - Triple Encoding
+For triples, each node in the triple (subject, predicate, object) is individually encoded as a string and stored sequentially in the byte buffer.
+
+### Decoding Process
+#### Node Decoding Process
+1. **Byte Extraction**: The stored bytes are read and converted back to a string.
+2. **Node Type Identification**: The node type is determined from the string format.
+3. **String Decoding**: The string is decoded back to its original node form.
+4. **Triple Decoding**: The process is repeated for each node in the triple.
+
+#### Node - Byte Extraction
+The decoding process starts by extracting the byte array from the storage buffer. This byte array is then converted back into a string.
+Node Type Identification:
+
+The node type is determined by examining the prefix or marker in the string. For instance:
+- **"u:"** indicates a URI.
+- **"l:"** indicates a Literal.
+- **"b:"** indicates a Blank node.
+- **"*"** indicates an Any node.
+
+#### Node - String Decoding
+The string, now identified by its type, is decoded back into its original node form. This involves stripping off the type marker and reconstructing the node from the remaining string.
+
+#### Node - Triple Decoding
+The process is repeated for each node in a triple, resulting in the reconstruction of the original subject, predicate, and object nodes.
+
+---
+## Keys - By ID
+### Overview
+The [StoreFmtById class](../rdf-abac-core/src/main/java/io/telicent/jena/abac/labels/StoreFmtById.java)
+ uses an ID-based approach to convert nodes to Node-IDs and back again, optimizing storage efficiency. The IDs are generated by making use of Jena's Node Table class.
+
+The downside being that the table is held in memory and so upon restart - the table is no longer available and thus the IDs are potentially invalidated.
```

Comment:
Ok, so Jena definitely has persistent Node Tables's since they are used internally in TDB so maybe a ticket for the Backlog about how we could implement that in future.

Also for the case of the labels store being for a TDB database anyway might be possible to just reference the existing persistent Node Table somehow

Reply:
@rvesse I thought so put didn't investigate further than what was available here.
Agreed, the more options the better. I'll create a ticket as part of what I'm doing now.